### PR TITLE
Modernize Objc codebase

### LINKED ISSUE #2642: Modernize Objc codebase
## ⚒ Description
Our Objc codebase's style and structure seems to be out of date and there are bunch of smell code, which can slow us down when developing for long journey.

There are many best practice, which we can acquire easily to upgrade our codebase to new level.

This[ article from Ash Furrow ](https://ashfurrow.com/blog/structuring-modern-objective-c/) is one of the best resource, which explain carefully what the problem is and how we tackle it.

## ⭐️ Why do you want this?
We all want our codebase looks shiny and high quality, right? It's time to clean it up 🐞

## 🎩 Acceptance Criteria
1. Limit the accessibility of model's properties.
Examples:
+ There are a lot of public properties and they are writeable:
https://github.com/toggl/toggldesktop/blob/803d43a723ca36d562970a9ad9991a5bc0b0ee4c/src/ui/osx/TogglDesktop/TimeEntryViewItem.h#L15-L19
=> Use Read-only Properties in Headers

2. Should declare `nonatomic`
https://github.com/toggl/toggldesktop/blob/803d43a723ca36d562970a9ad9991a5bc0b0ee4c/src/ui/osx/TogglDesktop/TimeEntryViewItem.h#L29-L32
=> `atomic` leads to slow down the compile time and it doesn't provide full safe-thread ability.

3. `String` should be `copy` than `strong`
https://github.com/toggl/toggldesktop/blob/803d43a723ca36d562970a9ad9991a5bc0b0ee4c/src/ui/osx/TogglDesktop/TimeEntryViewItem.h#L58
=> `strong` mean the string is passed by reference. From my understand, it's the potential error, which leads to untraceable bugs.
=> `copy` is best practice in modern objc codebase, since it's copy value and `immutable`.

4. Fix warnings on Project and 3rd-parties projects

5. Don’t #import in Header Files Unnecessarily
There are a lot of benefits, such as faster compile time and avoid circular `#import` statement.
Ex: https://github.com/toggl/toggldesktop/blob/803d43a723ca36d562970a9ad9991a5bc0b0ee4c/src/ui/osx/TogglDesktop/TimeEntryListViewController.h#L10-L13

6. Move `#import "toggl_api.h"` to `precompiled header`.
https://github.com/toggl/toggldesktop/blob/803d43a723ca36d562970a9ad9991a5bc0b0ee4c/src/ui/osx/TogglDesktop/TimeEntryListViewController.h#L10

### PR DESCRIPTION
**Pull request by NghiaTranUIT**
### 📒 Description
It's time to refine the quality of Objc coodebase

### 🕶️ Types of changes
**Breaking change** (fix or feature that would cause existing functionality to change)

### 🤯 List of changes
- [x] Refactor UI objc codebase by following the best-practice, which is carefully described in #2642  
- [x] Refactor AutoComplete classes

### 👫 Relationships
Close #2642 

### 🔎 Review hints
- All basic logic are remained

### CODE REVIEW

File: src/ui/osx/TogglDesktop/test2/TogglDesktop-Prefix.pch [hunk 1]
```diff
@@ -5,3 +5,5 @@
 #ifdef __OBJC__
     #import <Cocoa/Cocoa.h>
 #endif
+
+#import "toggl_api.h"
```
Comment (NghiaTranUIT) [inline src/ui/osx/TogglDesktop/test2/TogglDesktop-Prefix.pch]: By adding `toggl_api.h` in pre-complied header, it's optimized the build-time and reduce the redundant of import statement

**Comment by NghiaTranUIT** [comment]
Updated: `TimeEntryEditViewController` and few unused classes didn't intentionally modernize it since it's unnecessary and they are going to be removed in https://github.com/toggl/toggldesktop/pull/3112

**Comment by NghiaTranUIT** [comment]
Rebased completely due to file conflicts from master. It's ready to review 👍

**Comment by skel35** [comment]
@NghiaTranUIT I think there are some files that were removed in #3112 (like `TimeEntryEditViewController.h`) and added back here when resolving the merge conflict. Can you check those files? The number of added lines also looks like some files were added
image

**Comment by NghiaTranUIT** [comment]
@skel35 thank for the point. It has been fixed 👍

**Comment by NghiaTranUIT** [comment]
@tanel All done 👍 Let proceed 💯

**Pull request merged**